dataflow/encryption-keys/README.md

```diff
@@ -0,0 +1,197 @@
+# Using customer-managed encryption keys
+
+[![Open in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/editor)
+
+This sample demonstrate how to use
+[cryptographic encryption keys](https://cloud.google.com/kms/)
+for the I/O connectors in an
+[Apache Beam](https://beam.apache.org) pipeline.
+For more information, see the
+[Using customer-managed encryption keys](https://cloud.google.com/dataflow/docs/guides/customer-managed-encryption-keys)
+docs page.
+
+## Before you begin
+
+Follow the
+[Getting started with Google Cloud Dataflow](../README.md)
+page, and make sure you have a Google Cloud project with billing enabled
+and a *service account JSON key* set up in your `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+Additionally, for this sample you need the following:
+
+1. [Enable the APIs](https://console.cloud.google.com/flows/enableapi?apiid=bigquery,cloudkms.googleapis.com):
+   BigQuery and Cloud KMS API.
+
+1. Create a Cloud Storage bucket.
+
+   ```sh
+   export BUCKET=your-gcs-bucket
+   gsutil mb gs://$BUCKET
+   ```
+
+1. [Create a symmetric key ring](https://cloud.google.com/kms/docs/creating-keys).
+   For best results, use a [regional location](https://cloud.google.com/kms/docs/locations).
+   This example uses a `global` key for simplicity.
+
+   ```sh
+   export KMS_KEYRING=samples-keyring
+   export KMS_KEY=samples-key
+
+   # Create a key ring.
+   gcloud kms keyrings create $KMS_KEYRING --location global
+
+   # Create a key.
+   gcloud kms keys create $KMS_KEY --location global \
+     --keyring $KMS_KEYRING --purpose encryption
+   ```
+
+   > *Note:* Although you can destroy the
+   > [*key version material*](https://cloud.google.com/kms/docs/destroy-restore),
+   > you [cannot delete keys and key rings](https://cloud.google.com/kms/docs/object-hierarchy#lifetime).
+   > Key rings and keys do not have billable costs or quota limitations,
+   > so their continued existence does not impact costs or production limits.
+
+1. Grant Encrypter/Decrypter permissions to the *Dataflow*, *Compute Engine*, and *BigQuery* accounts.
```

Comment:
We should try to do a better job explaining what this means.  Are there pages we can link to that explain this notion of "service accounts created by a service" vs those that are user created?  If not, can we at least indicate that's what's going on here, as just calling them "dataflow/compute/bigquery account" is ambiguous/confusing.

Reply:
Good point, I'll add some more descriptions and look out for any useful links as well.